hub-backend/src/main/java/org/panda_lang/hub/utils/entity/AbstractCrudController.kt

```diff
@@ -36,13 +44,13 @@ import javax.validation.Valid
  * @param <U> type of update dto
  * @param <C> type of create dto
  */
-abstract class AbstractCrudController<S : CrudOperationsService<T, ID>, T : IdentifiableEntity<ID>, ID, U : AbstractDto<T>, C : AbstractDto<T>>(
-        protected open val service: S
-) {
-
-    companion object {
-        private const val SPEL_EXPRESSION = "(isAuthenticated() && principal.user.identifier.equals(#id)) || hasAuthority('ADMIN')"
-    }
+abstract class AbstractCrudController<S, T, ID, U, C>(
+    protected open val service: S
+) where
+S : CrudOperationsService<T, ID>,
+    T : IdentifiableEntity<ID>,
+    U : AbstractDto<T>,
+    C : AbstractDto<T> {
```

Comment:
Kotlin has `where` constraits which do the same job as the former generic parameter declaration, but it's appended after the class signature in its own block, making the generic statement easier to grasp.